contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol

```diff
@@ -2,27 +2,101 @@
 pragma solidity ^0.8.19;
 
 import {IReceiver} from "./interfaces/IReceiver.sol";
+import {ConfirmedOwner} from "../shared/access/ConfirmedOwner.sol";
 
-contract KeystoneFeedsConsumer is IReceiver {
-  event MessageReceived(bytes32 indexed workflowId, address indexed workflowOwner, uint256 nReports);
-  event FeedReceived(bytes32 indexed feedId, uint256 price, uint64 timestamp);
+contract KeystoneFeedsConsumer is IReceiver, ConfirmedOwner {
+  event FeedReceived(bytes32 indexed feedId, int192 price, uint32 timestamp);
 
-  constructor() {}
+  error UnauthorizedSender(address sender);
+  error UnauthorizedWorkflowOwner(address workflowOwner);
+  error UnauthorizedWorkflowName(bytes10 workflowName);
 
-  struct FeedReport {
-    bytes32 FeedID;
-    uint256 Price;
-    uint64 Timestamp;
+  constructor() ConfirmedOwner(msg.sender) {}
+
+  struct ReceivedFeedReport {
+    bytes32 FeedId;
+    int192 Price;
+    uint32 Timestamp;
+  }
+
+  struct StoredFeedReport {
+    int192 Price;
+    uint32 Timestamp;
+  }
+
+  mapping(bytes32 feedId => StoredFeedReport feedReport) internal s_feedReports;
+  address[] internal s_allowedSendersList;
+  mapping(address => bool) internal s_allowedSenders;
+  address[] internal s_allowedWorkflowOwnersList;
+  mapping(address => bool) internal s_allowedWorkflowOwners;
+  bytes10[] internal s_allowedWorkflowNamesList;
+  mapping(bytes10 => bool) internal s_allowedWorkflowNames;
+
+  function setConfig(
```

Comment:
This is not efficient (I understand it is just an example). We can fix it in a follow-up gas optimization PR.